src/hotspot/share/gc/z/zCollectedHeap.cpp

```diff
@@ -105,11 +105,31 @@ class ZStopConcurrentGCThreadClosure : public ThreadClosure {
 
 void ZCollectedHeap::stop() {
   log_info_p(gc, exit)("Stopping ZGC");
+  log_gc_vtime();
   ZAbort::abort();
   ZStopConcurrentGCThreadClosure cl;
   gc_threads_do(&cl);
 }
 
+class ZVCPUThreadClosure : public ThreadClosure {
+private:
+  volatile jlong _vtime = 0;
+public:
+  virtual void do_thread(Thread *thread) {
+    if (thread->is_ConcurrentGC_thread() ||
+        strstr(thread->name(), "ZWorker") != nullptr) {
+      Atomic::add(&_vtime, os::thread_cpu_time(thread));
+    }
+  }
```

Comment:
Why does this exclude threads like the `ZDirector` and other ZGC background threads? That thread seems to clearly be relevant to ZGC operation, doing so would make the measurement incomplete.
The change does not exclude e.g. some random G1 "director" threads either, even if they do not contribute much to the result.

Reply:
Both Parallel and G1 reuse the same gc-worker threads in `safepoint_workers()` for non-gc work, just fyi.

Reply:
FWIW; I confirmed with @stefank that `_runtime_workers` shoud not be accounted for GC CPU time for ZGC.